pkg/storage/stores/indexshipper/downloads/table.go

```diff
@@ -153,15 +153,7 @@ func (t *table) ForEach(ctx context.Context, userID string, callback index.ForEa
 		}
 
 		if indexSet.Err() != nil {
-			level.Error(util_log.WithContext(ctx, t.logger)).Log("msg", fmt.Sprintf("index set %s has some problem, cleaning it up", uid), "err", indexSet.Err())
-			if err := indexSet.DropAllDBs(); err != nil {
-				level.Error(t.logger).Log("msg", fmt.Sprintf("failed to cleanup broken index set %s", uid), "err", err)
-			}
-
-			t.indexSetsMtx.Lock()
-			delete(t.indexSets, userID)
```

Comment:
This was the cause of the page `LokiBoltdbShipperTablesDownloadFailed` that we saw yesterday, we were unexpectedly cleaning up user index set always instead of the one that was broken.